src/transformer/events/core/badge_awarded.php

```diff
@@ -0,0 +1,125 @@
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace src\transformer\events\core;
+
+
+use src\transformer\utils as utils;
+
+/**
+ * Transformer for the user created event.
+ *
+ * @param array $config The transformer config settings.
+ * @param \stdClass $event The event to be transformed.
+ * @return array
+ */
+
+
+function badge_awarded(array $config, \stdClass $event) {
+
+    global $CFG;
+    $repo = $config['repo'];
+    if (isset($event->objecttable) && isset($event->objectid)) {
+        $event_object = $repo->read_record_by_id($event->objecttable, $event->objectid);
+    } else {
+        $event_object = array();
+    }
+
+    $recipient = $repo->read_record_by_id('user', $event->relateduserid);
+    $course = $repo->read_record_by_id('course', $event->courseid);
+    $lang = utils\get_course_lang($course);
+    $badge = $repo->read_record_by_id('badge', $event->objectid);
+    $other = unserialize($event->other);
+    $issuedid = $other['badgeissuedid'];
+    $manual = $repo->read_record_by_id('badge_manual_award', $issuedid);
+    $awarder = $manual ? (utils\get_user($config, $repo->read_record_by_id('user', $manual->issuerid))) : 'System';
+    $badgetype = [1 => "Global", 2 => "Course"][$badge->type];
+    $course =  $badge->courseid ? $repo->read_record_by_id('course', $badge->courseid) : null;
+    
+    $statement = [[
+        'actor' => utils\get_user($config, $recipient),
+        'verb' => ['id' => 'https://w3id.org/xapi/tla/verbs/achieved',
+                   'display' => [
+                       'en-US' => 'Achieved'
+                   ]],
+        'object' => ['id' =>  $config['app_url'].'/badges/overview.php?id='.$event->objectid,
+                     'objectType' => 'Activity',
+                     'definition' => [
+                         'name' => [$lang => $badge->name],
+                         'description' => ['en-US' => $badge->description],
+                         'type' => 'https://xapi.edlm/profiles/edlm-lms/concepts/activity-types/badge',
+                         'extensions' => [
+                             'https://xapi.edlm/profiles/edlm-lms/v1/concepts/activity-extensions/badge-type' =>  $badgetype,
+                             'https://xapi.edlm/profiles/edlm-lms/v1/concepts/activity-extensions/badge-version' => $badge->version                             
+                         ]
+                     ],
+        ],
+        'result' => [
+            'success' => true,
+            'response' => $badge->message
+                ],
+        'context' => [
+            'language' => $lang,
+            'instructor' => $awarder,
+            'contextActivities' =>  [
+                'category' => ['id' => 'https://lms.edlm',
+                               'objectType' => 'Activity',
+                               'definition' => [
+                                   'name' => [
+                                       'en-US' => 'EDLM Moodle LMS'
```

Comment:
you may find cases of `en-US` in the example statements and profile, as a convention let's keep it to `en`